autoload/test/strategy.vim

```diff
@@ -87,8 +87,8 @@ function! test#strategy#vimterminal(cmd) abort
   let term_position = get(g:, 'test#vim#term_position', 'botright')
   execute term_position . ' new'
   call term_start(!s:Windows() ? ['/bin/sh', '-c', a:cmd] : ['cmd.exe', '/c', a:cmd], {'curwin': 1, 'term_name': a:cmd})
-  au BufLeave <buffer> wincmd p
-  nnoremap <buffer> <Enter> :q<CR>
+  au BufDelete <buffer> wincmd p " switch back to last window
```

Comment:
Bonus fix: we can now use `BufDelete` instead (like we already do for the `neovim` strategy) since we're actually deleting buffers. This was the reason `BufDelete` didn't work in the first place, as mentioned in #341.